### DISCUSSION TITLE
Hello Thyme!

### DISCUSSION THREAD

**Discussion by MacNc2023** [TheAssemblyArmada/Thyme#970] [Q&A] [answered]
hello. I'm a Zerohour user and modder who has been following your work. 

I'm writing to ask a few questions about your work. 
It would be my pleasure if you could answer them at your convenience.

Here are some of my questions 

1. is this Thyme project about fixing the numerous errors and bugs in zerohour?

2. is the end result of this project to be redistributed after fixing the errors and bugs? 

3. If yes, what is the estimated duration of the project? 

4. I can't help with this project, but I personally would like to fix and refactor the existing limited code in the thyme program, is that okay?

Game bug questions 

1. If I open a SkirmishScript in WorldBuilder and try to put a unit on a vehicle by setting the Edit Script and Edit Team, it's almost impossible. (Vehicles can be forced, but it's buggy, helicopters can't, only buildings) 

Do you know of any scripts that will let you mount a vehicle?

2. Is there any way to fix the error of ChinaNukeLauncher moving in the Deploy state or repeating the deployment?

3. when constructing a building, is there any way to make it not build the building site first, but to build it when the Dozer arrives?

4. When repairing a building, is it possible to have multiple workers repair it?

*This article was written through a translator and may not read smoothly.

**Comment by xezon** [comment, marked as answer]
> is this Thyme project about fixing the numerous errors and bugs in zerohour?

Yes. In the main repository or a fork of it.

> is the end result of this project to be redistributed after fixing the errors and bugs?

Yes. This repository already compiles builds for download. And I expect that a completed Thyme will bundle with various other Projects/Mods such as [TheSuperHackers/GeneralsGamePatch](https://github.com/TheSuperHackers/GeneralsGamePatch).

> If yes, what is the estimated duration of the project?

Several years.

> Is there any way to fix the error of ChinaNukeLauncher moving in the Deploy state or repeating the deployment?

Some of these bugs were already fixed with the aforementioned project, other are left for Thyme to fix.

**Reply by MacNc2023** [reply]
TheSuperHackers/GeneralsGamePatch](https://github.com/TheSuperHackers/GeneralsGamePatch) 
Is this the file you've been working on?

I'd love to see a Zerohour with many of the errors fixed.

Is it possible to get a file of the work done so far? I'm not a developer, but as a tester, I'd be happy to help by reporting feedback, errors, issues, etc.

**Reply by xezon** [reply]
I work both on Thyme and GeneralsGamePatch, because these projects are complementary to each other in what they cover (game code + game data).

Compiled Thyme you can download from https://github.com/TheAssemblyArmada/Thyme/actions, click on top entry, then scroll down to Artifacts on next page and click on "thyme".

GeneralsGamePatch does not come as compiled entity right now. You would need to clone repository to local disk and run `BuildInstallRunWithGui.bat` script from https://github.com/TheSuperHackers/GeneralsGamePatch/tree/main/Patch104pZH/Scripts.

**Reply by MacNc2023** [reply]
Thanks Zexon. I tried two things as you pointed out yesterday. 

First, I put Thyme into the cloned ZH folder, and second, I ran the BuildInstallRunWithGui.bat file in the Script folder of GeneralsGamePatch_Main to install and run the element files.

Question, 
1. is Thyme an ErrorReporting program like WorldBuilder?

2-1. is the game run with BuildInstallRunWithGui.bat from GeneralsGamePatch the version you have been working on?

(The configuration files didn't seem to work together properly, so I had to put the files in GeneralsGamePatch-main\Patch104pZH\.Build\RawBundleItems inside the installed ZH folder separately to get the button images and unit shapes).

2-2. I'm not sure, but from my experience, GeneralsGamePatch seems to be linked to the last executable in the ZH installed on the C drive. 

Is there any way to change the path? 
It's linked to the ZH Orig.file, so I have to delete the GeneralsGamePatch again to run the Orig.file. 

2-3. I think it's because the contents of 2-1 are not handled properly, but when I test ChinaNukeLauncher, it moves in the deployment state and repeats the deployment frequently. 

2-4. The bugs don't seem to have changed, so it looks like my fixes didn't take effect. 

Can you test with the last modified file?


Nevertheless, I'm happy to test play with this. 
I'm amazed at the way you guys work and the scale.

**Reply by xezon** [reply]
You need to copy Thyme manually into the game directory. You also need the original game.dat to run the thyme.dll (Not Origin, Not EA App, Not First Decade, but CD version from 2003).

GeneralsGamePatch is build and installed through the build scripts we already mentioned. It works on all game installs.

**Reply by MacNc2023** [reply]
I'm sure you're busy, but thanks for the explanation.

I appreciate being able to get in touch with you.

**Reply by MacNc2023** [reply]
Hi Xezon. 

Thanks for the rework and fixing a 20+ year old zero hour error.

I'm not sure if this is the right place to ask this, but it's the only way I can get in touch with you, so I hope you understand if it's a problem. 

1. after copying Thyme to my zerohour directory, when I ran the exe file, a ThymeDebugLogFile file was created in my documentation zerohour directory. 

Is this the correct way to do it? 

Is there any way 

I have no idea what Thyme is used for, so if you have any screenshots or videos, I'd love to see them. 

2. is GeneralsGamePatch-main the version with the bugs fixed so far?  
I ran the Patch104pZH\Scripts\BuildInstallRunWithGui.bat file to Excute as you said, but the ChinaNukeLauncher still moves in the Deploy state, and there is still the bug of repeating the Deploy, so I'm wondering if I ran the patched version. 

<In your response>. 
You need to copy Thyme manually into the game directory. You also need the original game.dat to run the thyme.dll (Not Origin, Not EA App, Not First Decade, but CD version from 2003).

original game.dat You said I need this file, how can I get it? 

I know you're busy and tired, but thanks for giving us modders hope.
